github-actions/post-approval-changes/lib/main.ts

```diff
@@ -131,6 +110,25 @@ async function runPostApprovalChangesAction(client: Octokit): Promise<void> {
   });
 }
 
+/** Set of membership lookup results, used as cache for lookups. */
+const isGooglerOrgMemberCache = new Map<string, boolean>([]);
+
+async function isGooglerOrgMember(client: Octokit, username: string): Promise<boolean> {
+  if (isGooglerOrgMemberCache.has(username)) {
+    return isGooglerOrgMemberCache.get(username)!;
+  }
+  return await client.orgs
+    .checkMembershipForUser({org: 'googler', username})
```

Comment:
This means everybody needs to have the membership public. I suggested that initially, so I'm good with it.

Though, two questions:

* Do you want to make sure we use a token that has access to the organization, so that we can even see for Googlers having their visibility private? obviously some googlers might not be part of the org at all..
* Are we announcing the change if we are doing it?

Reply:
```suggestion
    .checkMembershipForUser({org: 'googlers', username})
```

Reply:
I can work on getting the token access for checking private membership in Googler org.  But for now we can rely on public.  And all Googlers are supposed to register internally which places them in that org, so it allows for us to have a nice proxy for checking the internal database.


We definitely should announce if it changes.

Reply:
Haha yes, I was suggesting that GitHub org initially: https://docs.google.com/document/d/1Ab91dcLx87fNmqeiSW-CurZVDdSHYl_P5gmgQ_jtEWE/edit?usp=drivesdk&resourcekey=0-k_m13wmNFURdP-1bbVX_pA&disco=AAAAjO1Vazg

Yes, we should try supporting a token here for private visibility & announce this because not everybody is public in that org.

Reply:
I have set up the access for our Github App on the `googlers` organization. So now we are able to check private and public membership.